### PR TITLE
[GH-1057] Highlighting on hover for property values in the card dialog

### LINKED ISSUE #1057: Feature Idea: Property value controls should highlight on hover in the card editor
**Summary:**
In the card editor, hovering over a property name highlights it, but not when hovering over the value. this makes it hard to select the control to edit a property. The value control should have a similar gray highlight on hover.

https://user-images.githubusercontent.com/46905241/130283892-60ae3cf6-8ff5-4dcd-a730-65e1402adf51.mov

**How important this is to me and why:**

Importance: Medium

**Additional context / similar features:**
This should be a simple change, with a relatively high boost to usability.

### PR DESCRIPTION
**Pull request by kamre**
#### Summary
Highlighting on hover for property values in the card dialog:
- only values that are _not readonly_ highlighted (similar to Notion)
- all fields for values have `min-width` equal to `150px` (same as for property name)
- text fields like Email/URL/... are expanded when needed
- width of property name buttons is the same
- `LastModifiedAt`/`LastModifiedBy`/`CreatedAt` are considered _readonly_

Here is how it looks:
HighlightOnHover

#### Ticket Link
Fixes #1057

@asaadmahmood please review

### CODE REVIEW

File: webapp/src/components/properties/link/link.scss [hunk 1]
```diff
@@ -14,9 +17,8 @@
...
     &:hover {
         a {
-            display: inline;
+            display: flex;
+            align-items: center;
         }
     }
 }
-
-
```
Comment (kamre) [inline webapp/src/components/properties/link/link.scss:22]: This file was taken from #982

File: webapp/src/widgets/editable.tsx [hunk 2]
```diff
@@ -115,9 +116,26 @@ export function useEditable(
...
 const Editable = (props: EditableProps, ref: React.Ref<Focusable>): JSX.Element => {
     const elementRef = useRef<HTMLInputElement>(null)
     const elementProps = useEditable(props, ref, elementRef)
+
+    useLayoutEffect(() => {
+        if (props.autoExpand && elementRef.current) {
+            const input = elementRef.current
+            const computed = getComputedStyle(input)
+            input.style.width = 'auto'
+            input.style.width = `${input.scrollWidth + borderWidth(computed) + 1}px`
```
Comment (kamre) [inline webapp/src/widgets/editable.tsx:135]: `+ 1` here is required for workaround [this Chrome issue](https://bugs.chromium.org/p/chromium/issues/detail?id=980476).

File: webapp/src/components/properties/dateRange/dateRange.scss [hunk 3]
```diff
@@ -1,11 +1,6 @@
 .DateRange {
-    width: 100%;
     display: flex;
 
-    .octo-propertyvalue {
-        white-space: none;
```
Comment (kamre) [inline webapp/src/components/properties/dateRange/dateRange.scss:6]: Wrong value for `white-space` property.

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
Works fine. @asaadmahmood can you run this and verify the UI/UX changes?

**Comment by kamre** [comment]
@asaadmahmood by the way changes here somewhat overlap with yours in #1047

**Comment by asaadmahmood** [comment]
@harshilsharma63 @kamre Will have a look at this today.

**Review by asaadmahmood — APPROVED** [review]
Looks good to me, pushed a change.

**Comment by kamre** [comment]
@chenilim do we need any additional changes for this pull request?

**Comment by chenilim** [comment]
Sorry for the delay @kamre! In general, we just need one dev review to merge. We'll tighten up our processes.

Also, I manually merged with main, so please LMK if I missed anything. Thanks again for your contributions! These really up the basic experience of Focalboard!

**Pull request merged**